sys/dev/gpio/gpioled.c

```diff
@@ -94,27 +93,101 @@ gpioled_probe(device_t dev)
 	return (BUS_PROBE_DEFAULT);
 }
 
+static int
+gpioled_inv(device_t dev, uint32_t *pin_flags)
+{
+	struct gpioled_softc *sc;
+	int invert;
+	uint32_t pin_caps;
+
+	sc = device_get_softc(dev);
+
+	if (resource_int_value(device_get_name(dev),
+	                       device_get_unit(dev), "invert", &invert))
+		invert = 0;
+
+	if (GPIOBUS_PIN_GETCAPS(sc->sc_busdev, sc->sc_dev, GPIOLED_PIN,
+	                        &pin_caps) != 0) {
+		if (bootverbose)
+			device_printf(sc->sc_dev, "unable to get pin caps\n");
+		return (-1);
+	}
+	if (pin_caps & GPIO_PIN_INVOUT)
+		*pin_flags &= ~GPIO_PIN_INVOUT;
+	sc->sc_softinvert = 0;
+	if (invert) {
+		const char *invmode;
+
+		if (resource_string_value(device_get_name(dev),
+		                          device_get_unit(dev), "invmode", &invmode))
+			invmode = NULL;
+
+		if (invmode) {
+			if (!strcmp(invmode, "sw"))
+				sc->sc_softinvert = 1;
+			else if (!strcmp(invmode, "hw")) {
+				if (pin_caps & GPIO_PIN_INVOUT)
+					*pin_flags |= GPIO_PIN_INVOUT;
+				else {
+					device_printf(sc->sc_dev, "hardware pin inversion not supported\n");
+					return (-1);
+				}
+			} else if (strcmp(invmode, "auto") != 0) {
+				device_printf(sc->sc_dev, "invalid pin inversion mode\n");
+				invmode = NULL;
+			}
```

Comment:
This doesn't work if `invmode` is "auto" since it just skips over the assignment.
It should be:
```suggestion
			} else {
				if (strcmp(invmode, "auto") != 0)
					device_printf(sc->sc_dev,
					    "invalid pin inversion mode\n");
				invmode = NULL;
			}
```